src/utils/hook.ts

```diff
@@ -55,6 +55,32 @@ export const useTimeoutWithUnmount = (callback: () => void, clearCallback: () =>
   }, [delay])
 }
 
+export function useBoolean(initialState: boolean): [
+  boolean,
+  {
+    on: () => void
+    off: () => void
+    toggle: (newState?: boolean) => void
+  },
+] {
+  const [state, setState] = useState(initialState)
+
+  const on = useCallback(() => setState(true), [])
+  const off = useCallback(() => setState(false), [])
+  const toggle = useCallback(newState => {
+    setState(oldState => (newState != null ? newState : !oldState))
```

Comment:
What if use `typeof newState === 'boolean'` for better type checking if type casting(`!!newState`) is not expected?

Reply:
The purpose of using TypeScript is to solve the mental burden of similar problems.
My expectation is that all internal flow types are as expected, and if they are not, the problem areas should be fixed, not here.

Reply:
LGTM